### PR TITLE
fixes #30499 - update latest versions of fedora, ruby and node

### PR DESCRIPTION
**Pull request by ohadlevy**
Also removed a warning which is visible with ruby 2.7 / rails 6, by using
```
bundle exec rails app:update:bin
bundle exec spring binstub
```

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
Can you elaborate on why spring is removed? No objection, but I'd like to understand.

**Comment by ohadlevy** [comment]
> Can you elaborate on why spring is removed? No objection, but I'd like to understand.

Its the outcome of running ```bundle exec rails app:update:bin``` some more context at https://stackoverflow.com/questions/36538089/what-is-the-purpose-of-railsupdatebin-rake-task

**Comment by ohadlevy** [comment]
is the test failure known? 
```
Expected {:name=>"_ForemanSelectedhosts", :value=>"%5B1%5D", :path=>"/", :domain=>"127.0.0.1", :expires=>nil, :secure=>false} to be empty. (Minitest::Assertion)
/home/jenkins/workspace/test_develop_pr_core/database/postgresql-integrations/ruby/2.5/slave/fast/test/integration/host_js_test.rb:401    
```

**Comment by ekohl** [comment]
https://github.com/rails/rails/commit/a6c958b3a7d851d6609188dd4faa5f98fa23f381 suggests this should be moved to https://github.com/theforeman/foreman/blob/develop/config/boot.rb

**Comment by ohadlevy** [comment]
thanks, updated.

**Comment by ohadlevy** [comment]
[test katello]

**Comment by ekohl** [comment]
> thanks, updated.

I don't see any changes to `config/boot.rb`.

**Comment by ohadlevy** [comment]
> > thanks, updated.
> 
> I don't see any changes to `config/boot.rb`.

looking at the latest version of https://github.com/rails/rails/blob/6-0-stable/railties/lib/rails/generators/rails/app/templates/config/boot.rb.tt I dont think its actually needed anymore?

**Comment by ekohl** [comment]
Odd, in master it is there: https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/config/boot.rb.tt

https://github.com/rails/rails/commit/a6c958b3a7d851d6609188dd4faa5f98fa23f381 doesn't appear to be in the 6.0 branch, nor any spring. The fact we have it here since 4b836b6295dd2e928ee452a514fef51432ce2b73 suggests we haven't been in line with upstream for ages. I'm not sure if removal is wise.

**Comment by ohadlevy** [comment]
> Odd, in master it is there: https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/config/boot.rb.tt
> 
> [rails/rails@a6c958b](https://github.com/rails/rails/commit/a6c958b3a7d851d6609188dd4faa5f98fa23f381) doesn't appear to be in the 6.0 branch, nor any spring. The fact we have it here since [4b836b6](https://github.com/theforeman/foreman/commit/4b836b6295dd2e928ee452a514fef51432ce2b73) suggests we haven't been in line with upstream for ages. I'm not sure if removal is wise.

OK, I've added it back, the code from masters seems solid, so I wouldn't worry about that (one less task for 6.1?)

File: config/boot.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
+load File.expand_path("../bin/spring", __dir__)
```
Comment (ekohl) [inline config/boot.rb]:
```suggestion
begin
  load File.expand_path("../bin/spring", __dir__)
```

**Review by ekohl — DISMISSED** [review]
@adamruzicka any concerns?

**Comment by ohadlevy** [comment]
I've reverted the spring changes, retesting the container...

**Comment by ohadlevy** [comment]
ok, reverting the spring change is breaking the containers all together:
```
bundler: failed to load command: bin/rails (bin/rails)
LoadError: cannot load such file -- bundler
  /home/foreman/bin/spring:8:in `require'
  /home/foreman/bin/spring:8:in `<top (required)>'
  bin/rails:3:in `load'
  bin/rails:3:in `<top (required)>'
foreman_app_1
```
I assume this can be reproduce with ruby 2.7 now?

**Comment by ekohl** [comment]
You should not ship `bin/spring` in a container since it's development only. Spring itself is only listed in `bundler.d/development.rb`. We don't ship it our packages either. Then it doesn't exist and isn't loaded.

**Comment by ohadlevy** [comment]
> You should not ship `bin/spring` in a container since it's development only. Spring itself is only listed in `bundler.d/development.rb`. We don't ship it our packages either. Then it doesn't exist and isn't loaded.

I would prefer not to delete files (and not follow rails for that matter), how about this?

**Review by ekohl — CHANGES_REQUESTED** [review]
You should really just drop `bin/spring` when building. That means it'll simply not load it. Probably around here:
https://github.com/theforeman/foreman/blob/c4c097fab09588a886da113e0c598120644f5d34/Dockerfile#L47-L49

**Comment by ohadlevy** [comment]
ok, leave the patch or just it in the dockerfile?

**Comment by ohadlevy** [comment]
@ekohl thanks, Ijust ended up reverting all of the spring changes and deleted it from the container.

how doesnt this break production env from source?

**Comment by ekohl** [comment]
Normally the spring gem is unavailable because it's listed here:
https://github.com/theforeman/foreman/blob/c4c097fab09588a886da113e0c598120644f5d34/bundler.d/development.rb#L21

In our packages we simply remove the file `bundler.d/development.rb` so we know it's not present. This means `Spring` can never be loaded. All the scripts should gracefully fall back maybe the spring binstub isn't that graceful. Dropping the whole binstub feels appropriate since the gem isn't present either.

**Comment by ohadlevy** [comment]
btw, the above fix doesnt work either, bin/rails still reference it...
```
bundler: failed to load command: bin/rails (bin/rails)
LoadError: cannot load such file -- bundler
  /home/foreman/bin/spring:8:in `require'
  /home/foreman/bin/spring:8:in `<top (required)>'
  bin/rails:3:in `load'
  bin/rails:3:in `<top (required)>'
```
feels like I should reintroduce my earlier patch?

**Comment by ohadlevy** [comment]
sorry, PBKAC, I can confirm this patch works.

**Review by ekohl — APPROVED** [review]

**Comment by tbrisker** [comment]
Thanks @ohadlevy and @ekohl !

**Pull request merged**